cardinal/ecs/world.go

```diff
@@ -565,20 +567,30 @@ func (w *World) StartGameLoop(ctx context.Context, loopInterval time.Duration) {
 		w.Logger.Warn().Msg("No systems registered.")
 	}
 	go func() {
+		w.isGameLoopRunning = true
+		wasEndGameLoopSignalReceived := false
```

Comment:
you can actually do away with the `wasEndGameLoopSignalReceived` bool entirely by using labels and breaking on the specific label:

```go
loop:
	for {
		select {
		default:
			break loop
		}
	}
```

Reply:
yeah makes sense.